nvflare/app_common/data_exchange/data_exchanger.py

```diff
@@ -165,6 +169,6 @@ def _send_reply(self, data: Any, req_id: str, timeout: Optional[float] = None) -
         """
         if self.pipe_handler is None:
             raise RuntimeError("PipeMonitor is not initialized.")
-        msg = Message.new_reply(topic=self._topic, data=data, req_msg_id=req_id)
+        msg = Message.new_reply(topic=topic, data=data, req_msg_id=req_id)
         has_been_read = self.pipe_handler.send_to_peer(msg, timeout)
```

Comment:
Where is the retry logic in case the send_to_peer failed? Maybe it should be built into the pipe_handler.

Reply:
it should, let's fix that in another PR.